### ISSUE TITLE
Does not render without viewbox attribute

### ISSUE THREAD

**Issue by DaSpors** [BigBadaboom/androidsvg#166]
Trying to render the EUR symbol as Icon to a Button, but that fails when there's no viewport attribute present.

Sample EUR SVG that does not render:
```
<svg xmlns="http://www.w3.org/2000/svg" width="430" height="432">
	<path style="fill:#000" d="M0,198L55,198C54,204 54,210 54,216C54,222 54,228 55,234H16L0,270H61C86,362 171.5,432 270,432C324.5,432 375.5,410.5 414,375.5L414,323.5C381,367.5 328.5,396 270,396C190,396 121.5,343 98.5,270H342L358,234H91C90,228 90,222 90,216C90,211 90,204 91,198H374L390,162H98.5C121,90 189,36 270,36C333,36 381,64.5 414,108L430,72.5C390,28 333,0 270,0C171,0 85.5,71 61,162H16Z"/>
</svg>
```

And here working version *with* viewport:
```
<svg xmlns="http://www.w3.org/2000/svg" width="430" height="432" viewBox="0 0 430 432">
	<path style="fill:#000" d="M0,198L55,198C54,204 54,210 54,216C54,222 54,228 55,234H16L0,270H61C86,362 171.5,432 270,432C324.5,432 375.5,410.5 414,375.5L414,323.5C381,367.5 328.5,396 270,396C190,396 121.5,343 98.5,270H342L358,234H91C90,228 90,222 90,216C90,211 90,204 91,198H374L390,162H98.5C121,90 189,36 270,36C333,36 381,64.5 414,108L430,72.5C390,28 333,0 270,0C171,0 85.5,71 61,162H16Z"/>
</svg>
```

This is the code used for setting the icon:
```
public boolean setIcon(TextView tv, String svg_code, int icon_size)
    {
        try { // tested with icon_sizes 32, 48, 128

            SVG svg = SVG.getFromString(svg_code);
            svg.setDocumentWidth(icon_size);
            svg.setDocumentHeight(icon_size);
            Picture pic = svg.renderToPicture();

            PictureDrawable icon = new PictureDrawable(pic);
            icon.setBounds(0,0,icon_size,icon_size);
            tv.setCompoundDrawables(icon,null,null,null);

            return true;
        }
        catch (SVGParseException e)
        {
            CNP.Log(e);
        }
        return false;
    }
```

**Comment by BigBadaboom** [comment]
I believe your problem here is these two lines:

    svg.setDocumentWidth(icon_size);
    svg.setDocumentHeight(icon_size);

Your SVG is 430x432.  By setting the width and height to a small amount like 32 or 48, you will be ending up trying to render the blank area at the top left of your Euro symbol.

See this demo fiddle: https://jsfiddle.net/xrs8ewv0/

The green area represents the part of the SVG you will be drawing if you set `icon_size` to `48`.

Try something like the following instead:

            SVG svg = SVG.getFromString(svg_code);
            Picture pic = svg.renderToPicture(svg.getDocumentWidth(), svg.getDocumentHeight());

            PictureDrawable icon = new PictureDrawable(pic);
            ...etc...

In other words, don't attempt to change the document `width` and `height`.

This will only work if any SVG you are trying to draw has the `width` and `height` attributes.  You will get a NPE exception if they don't.

A better solution would be to make sure your SVGs have a viewBox.  Then you wouldn't have to worry about all this.

**Comment by DaSpors** [comment]
Thanks for the detailed analysis and documentation! In fact i already placed the viewport in the SVG (about 200 files). 

Will try your suggestion because I cannot guarantee that in future all files will contain a viewport.
